File: base-idm/idm-sdk/src/main/java/com/idm/sdk/model/OauthClientDetails.java
package com.idm.sdk.model;

import java.io.Serializable;


/**
 * The DTO class for the IDM_OAUTH_CLIENT_DETAILS database table.
 * 
 * @author Hafidz Malik
 * @since Dec 5, 2019
 */
public class OauthClientDetails implements Serializable {
	

	private static final long serialVersionUID = -6474824992620987069L;

	private String clientId;

	private Integer accessTokenValidity;

	private String additionalInformation;

	private String authorities;

	private String authorizedGrantTypes;

	private String autoapprove;

	private String clientSecret;

	private Integer refreshTokenValidity;

	private String resourceIds;

	private String scope;

	private String webServerRedirectUri;

	
	public OauthClientDetails() {
		// do nothing
	}

	public String getClientId() {
		return this.clientId;
	}

	public void setClientId(String clientId) {
		this.clientId = clientId;
	}

	public Integer getAccessTokenValidity() {
		return this.accessTokenValidity;
	}

	public void setAccessTokenValidity(Integer accessTokenValidity) {
		this.accessTokenValidity = accessTokenValidity;
	}

	public String getAdditionalInformation() {
		return this.additionalInformation;
	}

	public void setAdditionalInformation(String additionalInformation) {
		this.additionalInformation = additionalInformation;
	}

	public String getAuthorities() {
		return this.authorities;
	}

	public void setAuthorities(String authorities) {
		this.authorities = authorities;
	}

	public String getAuthorizedGrantTypes() {
		return this.authorizedGrantTypes;
	}

	public void setAuthorizedGrantTypes(String authorizedGrantTypes) {
		this.authorizedGrantTypes = authorizedGrantTypes;
	}

	public String getAutoapprove() {
		return this.autoapprove;
	}

	public void setAutoapprove(String autoapprove) {
		this.autoapprove = autoapprove;
	}

	public String getClientSecret() {
		return this.clientSecret;
	}

	public void setClientSecret(String clientSecret) {
		this.clientSecret = clientSecret;
	}

	public Integer getRefreshTokenValidity() {
		return this.refreshTokenValidity;
	}

	public void setRefreshTokenValidity(Integer refreshTokenValidity) {
		this.refreshTokenValidity = refreshTokenValidity;
	}

	public String getResourceIds() {
		return this.resourceIds;
	}

	public void setResourceIds(String resourceIds) {
		this.resourceIds = resourceIds;
	}

	public String getScope() {
		return this.scope;
	}

	public void setScope(String scope) {
		this.scope = scope;
	}

	public String getWebServerRedirectUri() {
		return this.webServerRedirectUri;
	}

	public void setWebServerRedirectUri(String webServerRedirectUri) {
		this.webServerRedirectUri = webServerRedirectUri;
	}

}

File: angular/base-bff-api/projects/bff-api/src/lib/models/oauthClientDetails.ts

export class OauthClientDetails {
  clientId: string;
  accessTokenValidity?: number;
  additionalInformation?: string;
  authorities?: string;
  authorizedGrantTypes?: string;
  autoapprove?: string;
  clientSecret?: string;
  refreshTokenValidity?: number;
  resourceIds?: string;
  scope?: string;
  webServerRedirectUri?: string;

    constructor(args: {
      clientId: string,
      accessTokenValidity?: number,
      additionalInformation?: string,
      authorities?: string,
      authorizedGrantTypes?: string,
      autoapprove?: string,
      clientSecret?: string,
      refreshTokenValidity?: number,
      resourceIds?: string,
      scope?: string,
      webServerRedirectUri?: string
    }) {
      this.clientId = args.clientId;
      this.accessTokenValidity = args.accessTokenValidity;
      this.additionalInformation = args.additionalInformation;
      this.authorities = args.authorities;
      this.authorizedGrantTypes = args.authorizedGrantTypes;
      this.autoapprove = args.autoapprove;
      this.clientSecret = args.clientSecret;
      this.refreshTokenValidity = args.refreshTokenValidity;
      this.resourceIds = args.resourceIds;
      this.scope = args.scope;
      this.webServerRedirectUri = args.webServerRedirectUri;
   }
}
